File: CS502017/pset1/greedy/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float m;
    int i;
    int c;
    i=0;
    printf("O hai! How much change is owed?\n");
    m = get_float();
    c = round(m*100);
    while (m<0)
    {
        printf("How much change is owed?\n");
        m = get_float();
    }
    while (c>=25)
            {
                c=c-25;
                i++;
            }
    while (c>=10)
            {
                c=c-10;
                i++;
            }
    while (c>=5)
            {
                c=c-5;
                i++;
            }
    while (c>=1)
            {
                c=c-1;
                i++;
            }
        printf("%i\n",i);

}

File: CS502017/pset6/greedy.py
import cs50

print("O hai! How much change is owed?")

coins = 0

while True:
    amount = cs50.get_float()
    if (amount >= 0):
        break
    print("How much change is owed?")

amount = round(amount * 100)

while True:
    if amount >= 25:
        amount -= 25
        coins += 1

    elif amount >= 10:
        amount -= 10
        coins += 1

    elif amount >= 5:
        amount -= 5
        coins += 1

    elif amount >= 1:
        amount -= 1
        coins += 1

    elif amount == 0.0:
        print("{}".format(coins))
        break